File: java/gpt/P9.java
// Task p9: Takes an index argument and prints the corresponding argv argument
public class P9 {
    public static void main(String[] args) {
        if (args.length < 1) {
            System.out.println("Usage: java P9 INDEX [args...]");
            System.exit(0);
        }

        int idx;
        try {
            idx = Integer.parseInt(args[0]);
        } catch (NumberFormatException e) {
            System.out.println("Invalid index");
            System.exit(0);
            return;
        }

        int target = idx + 1;
        if (target >= 0 && target < args.length + 1) {
            // args array starts at args[0] which corresponds to argv[1] in C; to get the user-supplied arg we check args[target-1]
            if (target - 1 >= 0 && target - 1 < args.length) {
                System.out.println(args[target - 1]);
            } else {
                System.out.println("");
            }
        } else {
            System.out.println("");
        }

        System.exit(0);
    }
}

File: python/gpt/p9.py
# Task p9: Program takes an index argument and prints the corresponding argv entry
import sys


def main():
    if len(sys.argv) < 2:
        print("Usage: python p9.py INDEX [args...]")
        return 0

    try:
        idx = int(sys.argv[1])
    except ValueError:
        print("Invalid index")
        return 0

    # According to spec, when called as: ./argv_index 1 arg1 arg2 ...
    # the program should print the argument at position (idx + 1) in argv,
    # which is the first user-supplied argument when idx==1.
    target = idx + 1
    if 0 <= target < len(sys.argv):
        print(sys.argv[target])
    else:
        print("")

    return 0


if __name__ == "__main__":
    sys.exit(main())
